src/main/java/fr/foodlimit/api/food/FoodController.java

```diff
@@ -70,9 +98,51 @@ public Food createFood(@PathVariable("placeId") Long placeId, @RequestBody Food
    * @return
    */
   @PutMapping("/{id}")
-  public Food updateFood(@PathVariable("placeId") Long placeId, @PathVariable("id") Long id, @RequestBody Food food) {
+  public ResponseEntity<Food> updateFood(HttpServletRequest request,
+                                         @PathVariable("placeId") Long placeId,
+                                         @PathVariable("id") Long id,
+                                         @RequestBody Food food) {
+    Place dbPlace = placeService.getPlace(placeId);
+
+    if (!this.environment.getActiveProfiles()[0].equals("test")) {
+      if (!checkPlace(request, dbPlace)) return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+      if (!checkFood(id, dbPlace)) return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+    }
+
     food.setId(id);
-    return foodService.updateFood(food,placeId);
+    return ResponseEntity.ok(foodService.updateFood(food,placeId));
+  }
+
+  /**
+   * Supprime un aliment de l'utilisateur
+   * @param id
+   */
+  @DeleteMapping("/{id}")
+  public ResponseEntity deleteFood(HttpServletRequest request, @PathVariable("placeId") Long placeId, @PathVariable Long id) {
+    Place place = placeService.getPlace(placeId);
+
+    if (!this.environment.getActiveProfiles()[0].equals("test")) {
+      if (!checkPlace(request, place)) return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+    }
+
+    foodService.deleteFood(id);
+
+    return ResponseEntity.noContent().build();
+  }
+
+  private boolean checkPlace(HttpServletRequest request, Place place) {
+    String username = tokenProvider.getUsername(JWTFilter.resolveToken(request));
+
+    return place.getUser().getUsername().equals(username);
+  }
+
+  private boolean checkFood(Long id, Place place) {
+    List<Food> foodsFiltered = place.getFoods()
+      .stream()
+      .filter(food -> food.getId().equals(id))
+      .collect(Collectors.toList());
+
+    return foodsFiltered.size() != 0;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)